src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -114,6 +115,11 @@ function createChainableTypeChecker(validate) {
     componentName = componentName || ANONYMOUS;
     propFullName = propFullName || propName;
     if (props[propName] == null) {
+      if (__DEV__) {
+        ReactInstrumentation.debugTool.onMissingProperty(componentName,
```

Comment:
This event seems awfully specific.  By firing more general events, we are able to expose an API that can be used for all sorts of devtools (even use cases we haven't though of yet).  I think I'd rather have an event that fires `onCreateChainableTypeChecker`.  The `if (props[propName] == null) {` can be replicated in the devtool.